### PR TITLE
Update doc elements to pass validate-modules test, remove usage of distutils.spawn

### PR DESCRIPTION
**Pull request by csmart**
This PR also merges in the changes from #114.

It fixes two issues relating to failing pipelines

### validate-modules

ansible-test's validate-modules sanity test was extended to also
validate the documentation (and thus configuration) for (documentable)
plugins. This introduced a schema which required updating of
documentation in order for the test to pass.

See https://github.com/ansible/ansible/pull/71734
See https://github.com/ansible-collections/news-for-maintainers/issues/9

Failing test:

```
$ ansible-test sanity --test validate-modules
Running sanity test "validate-modules"
ERROR: Found 9 validate-modules issue(s) which need to be resolved:
ERROR: plugins/connection/libvirt_lxc.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Michael Scherer <***>']
ERROR: plugins/connection/libvirt_lxc.py:0:0: invalid-documentation: DOCUMENTATION.connection: extra keys not allowed @ data['connection']. Got 'community.libvirt.libvirt_lxc'
ERROR: plugins/connection/libvirt_lxc.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
ERROR: plugins/connection/libvirt_qemu.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Jesse Pretorius <***>']
ERROR: plugins/connection/libvirt_qemu.py:0:0: invalid-documentation: DOCUMENTATION.connection: extra keys not allowed @ data['connection']. Got 'community.libvirt.libvirt_qemu'
ERROR: plugins/connection/libvirt_qemu.py:0:0: invalid-documentation: DOCUMENTATION.name: required key not provided @ data['name']. Got None
ERROR: plugins/inventory/libvirt.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Dave Olsthoorn <***>']
ERROR: plugins/inventory/libvirt.py:0:0: invalid-documentation: DOCUMENTATION.plugin_type: extra keys not allowed @ data['plugin_type']. Got 'inventory'
ERROR: plugins/inventory/libvirt.py:0:0: missing-gplv3-license: GPLv3 license header not found in the first 20 lines of the module
See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/validate-modules.html
ERROR: The 1 sanity test(s) listed below (out of 1) failed. See error output above for details.
validate-modules
```

### distutils

This is also merged with https://github.com/ansible-collections/community.libvirt/pull/114 

distutils has been deprecated and will be removed from Python's stdlib
in Python 3.12 (see https://www.python.org/dev/peps/pep-0632/). This PR
replaces the use of deprecated ``distutils.spawn.find_executable`` with
Ansible's ``get_bin_path`` in ``_search_executable`` function.

See https://github.com/ansible-collections/overview/issues/45#issuecomment-999911221

### CODE REVIEW

**Comment by csmart** [comment]
Note that the tests fail due to `distutils` check, therefore the separate PR #114 with a fix for that issue has been merged with this PR in order to combine the two changes and ensure tests are passing before merge is completed.

File: plugins/inventory/libvirt.py
Comment (csmart) [inline plugins/inventory/libvirt.py:6]: @odyssey4me do you mind checking whether me removing `plugin_type` is the right thing to do here? Thanks!

File: plugins/connection/libvirt_qemu.py
Comment (csmart) [inline plugins/connection/libvirt_qemu.py:15]: @odyssey4me I looked a few other collections and they don't seem to use the full name here, but I'd like to check if you think that this is the right thing to do. Thanks!

File: plugins/inventory/libvirt.py
Comment (odyssey4me) [reply plugins/inventory/libvirt.py:6]: Best to get advice from the core developers on this one. Perhaps something has changed, but this was originally a requirement - maybe they've reduced the friction a little bit.

File: plugins/connection/libvirt_qemu.py
Comment (odyssey4me) [reply plugins/connection/libvirt_qemu.py:15]: We originally had this as just the short name, but then it didn't work for someone and it got changed to the long, fully qualified name. I think we should get an answer from the Ansible core/community team here, rather than flip/flop this value.

Comment (csmart) [reply plugins/connection/libvirt_qemu.py:15]: @odyssey4me OK, I'll reach out there. Thanks!

File: plugins/inventory/libvirt.py
Comment (csmart) [reply plugins/inventory/libvirt.py:6]: Will, do thanks @odyssey4me!

File: plugins/connection/libvirt_qemu.py
Comment (csmart) [reply plugins/connection/libvirt_qemu.py:15]: This seems to have been addressed here https://github.com/ansible-collections/news-for-maintainers/discussions/11#discussioncomment-2299563 with @felixfontein confirming:

> name: for plugins is the same as module: for modules. It's the 'short name', not a FQCN.

File: plugins/inventory/libvirt.py
Comment (csmart) [reply plugins/inventory/libvirt.py:6]: And removal of `plugin_type` is confirmed here https://github.com/ansible-collections/news-for-maintainers/discussions/11#discussioncomment-2296426

**Comment by csmart** [comment]
Also thanks to @evgeni for confirming on Matrix that the changes are correct and same as `foreman` https://github.com/theforeman/foreman-ansible-modules/pull/1354/files

@odyssey4me I think this means the changes are OK, what do you think?

**Comment by evgeni** [comment]
FWIW, the diff looks absolutely sane to me :)

File: plugins/connection/libvirt_qemu.py
Comment (felixfontein) [inline plugins/connection/libvirt_qemu.py:14]: BTW, if you want to keep the email address, you can do:
```suggestion
    - Jesse Pretorius (@odyssey4me) <jesse@odyssey4.me>
```

Comment (csmart) [reply plugins/connection/libvirt_qemu.py]: Great, thanks @felixfontein! I have updated this to include the email.

**Review by Andersson007 — APPROVED** [review]

**Pull request merged**
